lib/resty/iprepd.lua

```diff
@@ -83,6 +96,21 @@ function _M.check(self, ip)
     -- return 403 and log rejections
     ngx.log(ngx.ERR, ip .. ' rejected with a reputation of ' .. reputation)
     ngx.exit(ngx.HTTP_FORBIDDEN)
+    if self.statsd then
+      self.statsd.incr("iprepd.status.rejected")
+    end
+  else
+    if self.statsd then
+      self.statsd.incr("iprepd.status.accepted")
+    end
+  end
+end
+
+function _M.flush_stats(self)
+  if self.statsd then
+    if #self.statsd.buffer >= self.statsd_buffer_size then
```

Comment:
The tables have no internal item count and `#` calculates item count each time

There might be some small performance benefit in including an item count with the buffer and just checking the value of that instead of calculating the table size, but it might not be worth it.